Userland/Libraries/LibWeb/Layout/SVGSVGBox.cpp

```diff
@@ -24,70 +24,60 @@ JS::GCPtr<Painting::Paintable> SVGSVGBox::create_paintable() const
-            return;
-        }
+    // https://www.w3.org/TR/SVG2/coords.html#SizingSVGInCSS
+
+    // The intrinsic dimensions must also be determined from the width and height sizing properties.
+    // If either width or height are not specified, the used value is the initial value 'auto'.
+    // 'auto' and percentage lengths must not be used to determine an intrinsic width or intrinsic height.
+    auto const& computed_width = computed_values().width();
+    if (computed_width.is_length() && !computed_width.contains_percentage()) {
+        set_intrinsic_width(computed_width.to_px(*this, 0));
     }
 
-    Optional<CSSPixelRect> united_rect;
-
-    auto add_to_united_rect = [&](CSSPixelRect const& rect) {
-        if (united_rect.has_value())
-            united_rect = united_rect->united(rect);
-        else
-            united_rect = rect;
-    };
-
-    for_each_in_subtree_of_type<SVGGeometryBox>([&](SVGGeometryBox const& geometry_box) {
-        auto& dom_node = const_cast<SVG::SVGGeometryElement&>(geometry_box.dom_node());
-        if (dom_node.has_attribute(HTML::AttributeNames::width) && dom_node.has_attribute(HTML::AttributeNames::height)) {
-            CSSPixelRect rect;
-            // FIXME: Allow for relative lengths here
-            rect.set_width(computed_values().width().to_px(*this, 0));
-            rect.set_height(computed_values().height().to_px(*this, 0));
-            add_to_united_rect(rect);
-            return IterationDecision::Continue;
-        }
+    auto const& computed_height = computed_values().height();
+    if (computed_height.is_length() && !computed_height.contains_percentage()) {
+        set_intrinsic_height(computed_height.to_px(*this, 0));
+    }
+
+    set_intrinsic_aspect_ratio(calculate_intrinsic_aspect_ratio());
+}
 
-        auto& path = dom_node.get_path();
-        auto path_bounding_box = path.bounding_box().to_type<CSSPixels>();
+Optional<float> SVGSVGBox::calculate_intrinsic_aspect_ratio() const
+{
+    // https://www.w3.org/TR/SVG2/coords.html#SizingSVGInCSS
+    // The intrinsic aspect ratio must be calculated using the following algorithm. If the algorithm returns null, then there is no intrinsic aspect ratio.
 
-        // Stroke increases the path's size by stroke_width/2 per side.
-        auto stroke_width = geometry_box.dom_node().stroke_width().value_or(0);
-        path_bounding_box.inflate(stroke_width, stroke_width);
+    auto const& computed_width = computed_values().width();
+    auto const& computed_height = computed_values().height();
 
-        auto& maybe_view_box = this->dom_node().view_box();
+    // 1. If the width and height sizing properties on the ‘svg’ element are both absolute values:
+    if (computed_width.is_length() && !computed_width.contains_percentage() && computed_height.is_length() && !computed_height.contains_percentage()) {
+        auto width = computed_width.to_px(*this, 0);
+        auto height = computed_width.to_px(*this, 0);
```

Comment:
Small copy & paste error:

```suggestion
        auto height = computed_height.to_px(*this, 0);
```